packages/expo-linear-gradient/src/normalizeColor.ts

```diff
@@ -5,23 +5,23 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */
-import { processColor } from 'react-native';
+import { ColorValue, processColor } from 'react-native';
 
 const isWebColor = (color: string): boolean =>
   color === 'currentcolor' ||
   color === 'currentColor' ||
   color === 'inherit' ||
   color.indexOf('var(') === 0;
 
-export function normalizeColor(color?: number | string, opacity: number = 1): void | string {
+export function normalizeColor(color?: ColorValue, opacity: number = 1): void | string {
   if (color == null) return;
 
   if (typeof color === 'string' && isWebColor(color)) {
     return color;
   }
 
-  const colorInt = processColor(color) as number | undefined;
-  if (colorInt != null) {
+  const colorInt = processColor(color);
+  if (typeof colorInt === 'number') {
```

Comment:
does this mean platform colors won't work on web? could we also handle a case where this is a string (?)

Reply:
They were never supported: https://github.com/necolas/react-native-web/issues/2128

In expo projects, the PlatformColor object is available in web builds but doesn't have any web-specific behaviour.

Anyhow, `processColor` either returns undefined, or a number, or an `OpaqueColorObject` which only contains platform-specific information like an asset catalog or system variable reference.

For web, the API consumer has to use `Platform.select(...) / Platform.OS === "web"` to resort to a color that returns a number here.